powershell-adapter/psDscAdapter/psDscAdapter.psm1

```diff
@@ -37,99 +37,197 @@ function Invoke-DscCacheRefresh {
         $Module
     )
 
-    # create a list object to store cache of Get-DscResource
-    [dscResourceCache[]]$dscResourceCache = [System.Collections.Generic.List[Object]]::new()
-
-    # improve by performance by having the option to only get details for named modules
-    # workaround for File and SignatureValidation resources that ship in Windows
-    if ($null -ne $module -and 'PSDesiredStateConfiguration' -ne $module) {
-        if ($module.gettype().name -eq 'string') {
-            $module = @($module)
+    $refreshCache = $false
+
+    $cacheFilePath = Join-Path $env:LocalAppData "dsc\PSAdapterCache.json"
+    if ($PSVersionTable.PSVersion.Major -le 5) {
+        $cacheFilePath = Join-Path $env:LocalAppData "dsc\WindowsPSAdapterCache.json"
+    }
+    if ($IsLinux -or $IsMacOS) {
+        $cacheFilePath = Join-Path $env:HOME "dsc" "PSAdapterCache.json"
+    }
+
+    if (Test-Path $cacheFilePath) {
+        $trace = @{'Debug' = "Reading from Get-DscResource cache file $cacheFilePath"} | ConvertTo-Json -Compress
```

Comment:
For all these traces, it would be better to have a `Write-Trace` helper function that does takes the string message and handles the JSON and writing to stderr

Reply:
Good idea; updated.